src/odemis/dataio/tiff.py

```diff
@@ -804,19 +807,13 @@ def _updateMDFromOME(root, das):
                 continue # might be a thumbnail, it's alright
             da.metadata.update(mdp)
 
-        # Convert the DeltaT's into PIXEL_DUR + TIME_OFFSET
+        # Update metadata of each da, so that they will be merged
+        nbchan = pos[1] + 1  # TODO think this is not the safest way to do it...
```

Comment:
Indeed, it could be almost safe if you were sure which key in pos is T.
I'd suggest to just check that the `len(deltats) == max(deltats.keys())`.